magefiles/run.go

```diff
@@ -0,0 +1,209 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+)
+
+// Create and run Kubernetes cluster with two nodes, Linux and Windows.
+func Run() error {
+	mg.SerialDeps(startup, checkVagrant, checkClusterNotRunning)
+
+	var err error
+
+	log.Println("Creating .lock directory")
+	_, err = os.Stat(".lock")
+	if !os.IsNotExist(err) {
```

Comment:
we should also fail if the ``.lock`` exists, so we don't run things twice at the same time, resulting in 2 clusters.

Also, ``if err == nil``, then ``os.IsNotExist(err)`` will be ``false``, which means that if we have the ``.lock``, we're going to remove the lock and create a new one. Is that the intent?

Reply:
Good point but this has already been resolve in the branch by pre-run check performed by internal target `checkClusterNotExist`